### PR TITLE
scripts: runners: improvements for spi_burn west runner

### PR DESCRIPTION
**Pull request by rikorsev**
Following changes has been made in this PR:
    - fix issue when a binary is not able to be found by relative path
    - implement "west debug" command
    - implement --dt-flash option

### CODE REVIEW

**Comment by rikorsev** [comment]
Hi @mbolivar-nordic,

May I ask you to review the PR?

Regards,
Alex

**Comment by rikorsev** [comment]
Hi @nashif, @kgugala, @pgielda,

Please, review the PR

Regards,
Alex

File: boards/common/spi_burn.board.cmake [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # SPDX-License-Identifier: Apache-2.0
 
 board_set_flasher_ifnset(spi_burn)
-board_finalize_runner_args(spi_burn)
+board_set_debugger_ifnset(spi_burn)
+board_finalize_runner_args(spi_burn)
```
Comment (mbolivar-nordic) [inline boards/common/spi_burn.board.cmake:5]: Please add missing newline at end of file.

File: boards/riscv/tlsr9518adk80d/doc/index.rst [hunk 2]
```diff
@@ -255,7 +255,23 @@ It is also possible to use the west flash command, but additional steps are requ
 Debugging
 =========
 
-Supporting UART debug and OpenOCD+GDB.
+Supporting UART debug and OpenOCD+GDB. west debug command also supported. You may run it in a simle way, like:
```
Comment (mbolivar-nordic) [inline boards/riscv/tlsr9518adk80d/doc/index.rst:258]:
```suggestion
This port supports UART debug and OpenOCD+GDB. The `west debug` command also supported. You may run it in a simple way, like:
```

**Review by mbolivar-nordic — COMMENTED** [review]
I'm sorry, I thought I had submitted my review and I lost track of this during the v3.1 release. Here are some initial comments.  Thanks for your patch.

File: boards/riscv/tlsr9518adk80d/doc/index.rst [hunk 3]
```diff
@@ -255,7 +255,20 @@ It is also possible to use the west flash command, but additional steps are requ
...
+   west debug
+
+Or with additional arguments, like:
+
+.. code-block:: console
+
+   west debug --gdb-port=<port_number> --gdb-ex=<additional_ex_arguments>
+
+Example:
+   west debug --gdb-port=1111 --gdb-ex="-ex monitor reset halt -ex b main -ex continue"
```
[the remark this answers is not in the document]
Comment (rikorsev) [reply boards/riscv/tlsr9518adk80d/doc/index.rst]: Done

File: scripts/west_commands/runners/spi_burn.py [hunk 4]
```diff
@@ -59,14 +71,49 @@ def do_run(self, command, **kwargs):
...
+
         # Start ICEman as background process
         cmd_ice_run = ["./ICEman", '-Z', 'v5', '-l', 'aice_sdp.cfg']
-        self.ice_process = subprocess.Popen(cmd_ice_run, stdout=subprocess.PIPE)
+
+        # Following part of code is necessary to ignore INT signal.
+        # which is necessary for correct working of GDB.
+        # The part is copied from popen_ignore_int. It is not possible
+        # to use popen_ignore_int directly due to abscense of possibility
+        # to pass stdout=subprocess.PIPE argument in Popen call
```
[the remark this answers is not in the document]
Comment (rikorsev) [reply scripts/west_commands/runners/spi_burn.py]: @mbolivar-nordic 
Thanks for good advice! I've been reworked start_iceman according to it. Please check.

File: boards/riscv/tlsr9518adk80d/doc/index.rst (hunk 2 above)
Comment (rikorsev) [reply boards/riscv/tlsr9518adk80d/doc/index.rst]: Done

**Comment by rikorsev** [comment]
Hi @mbolivar-nordic, @nashif, @kgugala, @pgielda,

Please, review the PR

Regards,
Alex

**Comment by fkokosinski** [comment]
Hi @rikorsev! The changes from `boards/riscv` look good. I'm not leaving an approval though, because the majority of your changes affect twister, so you will need to have this PR approved by its maintainers.

File: scripts/west_commands/runners/spi_burn.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mbolivar-nordic) [reply scripts/west_commands/runners/spi_burn.py]: Thanks for the update. The approach has a problem, however. You have preserved this commit as-is, then done the change I requested, then added a "fixup" commit. My original request was not to use a fixup commit, but instead:

> Please add a separate commit extending popen_ignore_int to take additional **kwargs which are passed on to the Popen constructor instead. Then in this commit, use the extended method, passing it any kwargs necessary.

That is:

1. rebase your second commit to be the first commit
2. squash the last commit into this commit, which will end up being the second commit

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Hi, zephyr policy is not to use "fixup" commits, so you need to rebase this pull request to adjust the commits as explained in more detail below.

**Comment by pgielda** [comment]
@rikorsev those changes seem to be related more to west than riscv, so you have to mostly discuss this with @mbolivar-nordic

**Comment by rikorsev** [comment]
Hi @mbolivar-nordic 

Please, review the PR and approve it if everything is fine.

Regards,
Alex

**Review by mbolivar-nordic — APPROVED** [review]

**Review by fkokosinski — APPROVED** [review]
LGTM for the RISC-V changes.

**Pull request merged**